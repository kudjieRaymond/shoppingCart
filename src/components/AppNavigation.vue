<template>
  <span>
    <v-navigation-drawer app v-model="drawer" class="brown lighten-2" dark disable-resize-watcher>
      <v-list>
        <v-list-item v-for="(item) in items" :key="item.title" link>
          <v-list-item-content>
            <v-list-item-title>{{ item.title }}</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list>
    </v-navigation-drawer>
    <v-app-bar color="brown darken-4" dark>
      <v-app-bar-nav-icon class="hidden-md-and-up" @click="drawer = !drawer"></v-app-bar-nav-icon>
      <v-toolbar-title>{{appTitle}}</v-toolbar-title>
      <div class="flex-grow-1"></div>
      <div v-if="!isAuthenticated" class="hidden-sm-and-down">
        <v-btn to="/login">SIGN IN</v-btn>
        <v-btn color="brown lighten-3" to="/register">REGISTER</v-btn>
      </div>
      <div v-else>
        <v-btn outlined color="white" @click="logout">Logout</v-btn>
      </div>
    </v-app-bar>
  </span>
</template>

<script>
export default {
  name: 'AppNavigation',
  data() {
    return {
      appTitle: 'Buy-Something',
      drawer: false,
      items: [{ title: 'Menu' }, { title: 'Sign In' }, { title: 'Register' }]
      //isAuthenticated: false
    };
  },
  computed: {
    isAuthenticated() {
      return this.$store.getters['auth/isAuthenticated'];
    }
  },
  methods: {
    logout() {
      this.$store.dispatch('auth/userSignOut');
    }
  }
};
</script>

<style scoped>
</style>