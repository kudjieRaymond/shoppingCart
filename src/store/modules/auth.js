//import { firebaseAuth } from '../../config/firebase';
import firebase from 'firebase';
import router from '@/routes';

const auth = {
  //namespaced: true,

  state: {
    isAuthenticated: false,
    user: null
  },
  mutations: {
    setUser(state, payload) {
      state.user = payload;
    },
    setIsAuthenticated(state, payload) {
      state.isAuthenticated = payload;
    }
  },
  actions: {
    userRegister({ commit }, { email, password }) {
      firebase
        .auth()
        .createUserWithEmailAndPassword(email, password)
        .then(() => {
          // commit('setUser', user);
          // commit('setIsAuthenticated', true);
          router.push({ name: 'homePage' });
        })
        .catch(() =>{
          commit('setUser', null);
          commit('setIsAuthenticated', false);
          throw new Error('Credential incorrect');
        });
    },
    userLogin(_, { email, password }) {
			
      firebase
        .auth()
        .signInWithEmailAndPassword(email, password)
        .then(() => {
          router.push({ name: 'homePage' });
        })
        .catch(() => {
          //this.error = err.message;
          throw new Error('Credential incorrect');
        });
    },
    userSignOut({ commit }) {
      firebase
        .auth()
        .signOut()
        .then(() => {
          commit('setUser', null);
          commit('setIsAuthenticated', false);
          //show message;
        })
        .catch(() => {
          commit('setUser', null);
          commit('setIsAuthenticated', false);
        });
    },
    fetchUser({ commit }, user) {
      //console.log(user !== null);
      commit('setIsAuthenticated', user !== null);
      if (user) {
				let  {email, emailVerified, uid} = user
        commit('setUser', {
          email,
          emailVerified,
          uid
        });
      } else {
        commit('setUser', null);
      }
    }
  },
  getters: {
    isAuthenticated(state) {
			//console.log(state.isAuthenticated);
      return state.user !== null && state.user !== undefined;
    },
    currentUser(state) {
      if (state && state.user) {
        return {
          email: state.user.email,
          emailVerified: state.user.emailVerified,
          uid: state.user.uid
        };
      } else {
        return {};
      }
    }
  }
};

export default auth;
