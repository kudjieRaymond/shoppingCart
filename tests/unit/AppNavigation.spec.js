import Vuetify from 'vuetify';
import { mount, createLocalVue } from '@vue/test-utils';
import AppNavigation from '@/components/AppNavigation.vue';
import Vuex from 'vuex'
import auth from '@/store/modules/auth';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('AppNavigation.vue', () => {
	let vuetify;
	let store;
	let state;

  beforeEach(() => {
		vuetify = new Vuetify();
		state= {
			isAuthenticated: true,
			user:{
				email:'johndoe@gmail.com'

			}
		}
		store = new Vuex.Store({
			modules:{
				auth:{
					state,
					getters:auth.getters

				}
			}
		})
  });
  it('show the name of the app', () => {
    //const title = 'Buy-Something';
    const wrapper = mount(AppNavigation, {
      localVue,
			vuetify,
			store
		});
		
			expect(wrapper.html()).toMatchSnapshot();

    //const title = wrapper.find('.v-toolbar-title');

    //expect(title.text()).toBe('Buy-Something');
    //const div = wrapper.find('div .v-toolbar__title');
    //expect(wrapper.contains('v-toolbar-title')).toBe(true);
  });
});
