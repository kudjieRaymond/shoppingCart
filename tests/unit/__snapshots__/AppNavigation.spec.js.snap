// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigation.vue show the name of the app 1`] = `
<span><v-navigation-drawer app="" dark="" disable-resize-watcher="" class="light-green darken-4"><v-list><v-list-item link=""><v-list-item-content><v-list-item-title>Menu</v-list-item-title></v-list-item-content></v-list-item><v-list-item link=""><v-list-item-content><v-list-item-title>Sign In</v-list-item-title></v-list-item-content></v-list-item><v-list-item link=""><v-list-item-content><v-list-item-title>Register</v-list-item-title></v-list-item-content></v-list-item></v-list></v-navigation-drawer> <v-app-bar color="light-green darken-4" dark=""><v-app-bar-nav-icon class="hidden-md-and-up"></v-app-bar-nav-icon> <v-toolbar-title><v-btn text="" to="/">Buy-Something</v-btn></v-toolbar-title> <div class="flex-grow-1"></div> <div><v-btn outlined="" color="white">Logout johndoe@gmail.com</v-btn></div> <div class="hidden-sm-and-down"><v-btn to="/cart" text="" color="white"><v-badge color="cyan" left=""> <v-icon large="" color="orange accent-3">shopping_cart</v-icon>
          (GHC )
        </v-badge></v-btn></div></v-app-bar></span>
`;
